pkg/transportrequest/gitutils.go

```diff
@@ -2,12 +2,52 @@ package transportrequest
 
 import (
 	"fmt"
+	pipergit "github.com/SAP/jenkins-library/pkg/git"
+	"github.com/SAP/jenkins-library/pkg/log"
 	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/pkg/errors"
+	"os"
 	"regexp"
 	"sort"
 )
 
+// needs to be replaced by mocks in the tests
+var getWorkDirectory = os.Getwd
+
+// FindIDInRange finds a ID according to the label in a commit range <from>..to.
+// We assume the git repo is present in the current working directory.
+func FindIDInRange(label, from, to string) (string, error) {
+	workdir, err := getWorkDirectory()
```

Comment:
the creation of the repo should be outside. findInCommits could be a member of the repo. 
And easier to mock.

Reply:
The reason why I did it like that is I would like to avoid having the same coding several times ...

On the other hand I think we will have more cases where we interact with the local repo which has already be materialized by the build environment (jenkins-git-plugin, azure dev-ops ...). Maybe we should provide that inside `pkg/git`. Will open another PR for that ...

Reply:
My sponaneous idea was to have a sort-of Walker pattern, like

func Walk(repo *git.Repository, from string, to string, cb func(c *object.Commit) error) {

Reply:
> Walker pattern

Don't we have something similar already?

- first we resolve `git diff start..end`, after that we traverse these commits and now we have to react on the the result:
  - there might be no label --> error
  - there might be one label --> we are happy (maybe the same label is contained in several commits)
  - there might be several labels --> error
 
I'm open for other approaches. @rodibrin : would you like to provide a sketch ?